### PR TITLE
More automated tests

### PR DESCRIPTION
**Pull request by puskin**
Added more automated tests, chipping in for the 100% test coverage goal mentioned in [here](https://github.com/fastify/busboy/issues/126)

I really don't know the project, if not for what I studied while writing tests, so I picked the least complicated route and wrote a couple of automated tests of "well exposed" functions and methods

### CODE REVIEW

**Review by gurgunday — APPROVED** [review]
lgtm

**Review by Uzlopak — CHANGES_REQUESTED** [review]
basename.test.js should cover all branches
busboy-emit.test.js should check if the first call to busboy.emit('finish') returns anything
dicer-write.test.js does not test what it claims

**Comment by puskin94** [comment]
> busboy-emit.test.js should check if the first call to busboy.emit('finish') returns anything

@Uzlopak am I reading the code wrong, or `Busyboy.emit` will never return anything?

File: test/dicer-write.test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  t.test('calls the callback cb() when headerFirst is set and all the data have been written', t => {
+    const dicer = new Dicer({ headerFirst: true })
+
+    dicer._write(Buffer.from('Content-Type: text/plain'), null, () => {
+      dicer._write(Buffer.from('Content-Type: text/plain'), null, () => {
+        t.pass('write method called')
+      })
+    })
+
+    t.end()
```
Comment (Uzlopak) [inline test/dicer-write.test.js:18]: Basically not testing anything, because you call t.end().  use t.plan for this subtest also.

Comment (puskin94) [reply test/dicer-write.test.js:19]: @Uzlopak oh, good to know! First time using this testing framework!

Comment (Uzlopak) [reply test/dicer-write.test.js:19]: remove the t.end() please

Comment (puskin94) [reply test/dicer-write.test.js]: @Uzlopak is there anything else for me to do in this PR?

**Review by Uzlopak — APPROVED** [review]

**Pull request merged**